Automattic/jetpack
# Boost: Add toggle UI with notice for Cache and trigger setup/cleanup

Part of https://github.com/Automattic/jetpack/issues/35054

I have intentionally not styled the retry button and have not added content for (some) errors, as we haven't finalized those. During our weekly meeting, we discussed the option to have the "run setup" button inside the notice. The current implementation, is just for debugging. Will be addressed in a follow up PR.

## Proposed changes:

* UI: Add a toggle to enable/disable the feature;
* Backend (PHP): Runs setup when feature is enabled;
* Backend (PHP): Runs cleanup when feature is disabled (cleans up advanced-cache.php and removes WP_CACHE from wp-config.php);
* UI: Add a notice that shows the last setup error. Maybe it would be a good idea to list all errors instead of just one?
* UI: Add a button to retry setup. Temporary for now, as we also need to add content for the various errors we might get during setup.
* React Data Sync Client: Add a way to invalidate queries. Used to invalidate the error data store, when the feature is enabled (to avoid showing the same error again);
* Backend (PHP): Add activate/deactivate traits, that are used when the feature is toggled on or off. This helps us run things on activation/deactivation (like setup and cleanup mentioned above). We should revisit existing code that runs code on activation/deactivation this in another way and clean it up.

<details><summary>Preview</summary>

![CleanShot 2024-02-13 at 14 06 05](https://github.com/Automattic/jetpack/assets/11799079/4743e030-2d7d-49bb-9187-572646b1cf9b)

</details>

 Since the feature is under a feature flag, don't forget to include `define( 'BOOST_CACHE', true );` in your `wp-config.php`.

### Other information:


## Jetpack product discussion



p1HpG7-r0o-p2

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






* Setup Boost;
* Make sure to add `define( 'BOOST_CACHE', true );` in your `wp-config.php` to get the toggle to show up;
* Make sure that the UI behaves as you'd expect it to. For example, if there's an error during setup, it should show. Fixing it and clicking the "re-run setup" should remove the notice or maybe show a different error.
